packages/lib/src/vis/scatter/ScatterVis.tsx

```diff
@@ -0,0 +1,77 @@
+import { assertDefined, ScaleType } from '@h5web/shared';
+import { Suspense } from 'react';
+
+import ColorBar from '../heatmap/ColorBar';
+import type { ColorMap } from '../heatmap/models';
+import { useAxisDomain, useDomain } from '../hooks';
+import PanZoomMesh from '../shared/PanZoomMesh';
+import VisCanvas from '../shared/VisCanvas';
+import { DEFAULT_DOMAIN } from '../utils';
+import ScatterPoints from './ScatterPoints';
+import styles from './ScatterVis.module.css';
+
+interface Props {
+  abscissas: number[];
+  ordinates: number[];
+  data: number[];
+  colorMap: ColorMap;
+  invertColorMap?: boolean;
+  scaleType?: ScaleType;
+  showGrid?: boolean;
+}
+
+function ScatterVis(props: Props) {
+  const {
+    abscissas,
+    ordinates,
```

Comment:
I'm a bit reluctant to calling these `abscissas` and `ordinates`, as these are the terms we use in other vis for arrays of regularly spaced, incremental or decremental axis values returned by `useAxisValues`. Not sure what else to call them, though ... maybe `scatteredAbscissas` and `scatteredOrdinates` ... oh, or what about `dataAbscissas` and `dataOrdinates` to make it clear that the arrays have the same length? On this note, it might be worth adding a couple of assertions to make sure the lengths of all three arrays match like we do with the `errors` array in `LineVis`.